application/controllers/FailedEmailController.php

```diff
@@ -0,0 +1,179 @@
+        ];
+        if (!$surveyId) {
+            throw new CHttpException(403, gT("Invalid survey ID"));
+        }
+        if (!$permissions['read']) {
+            App()->user->setFlash('error', gT("You do not have permission to access this page."));
+            $this->redirect(['surveyAdministration/view', 'surveyid' => $surveyId]);
+        }
+
+        App()->getClientScript()->registerScriptFile('/application/views/failedEmail/javascript/failedEmail.js', LSYii_ClientScript::POS_BEGIN);
+        $failedEmailModel = FailedEmail::model();
+        $failedEmailModel->setAttributes(App()->getRequest()->getParam('FailedEmail'), false);
+        $pageSize = App()->request->getParam('pageSize') ?? App()->user->getState('pageSize', App()->params['defaultPageSize']);
+        $massiveAction = App()->getController()->renderPartial('/failedEmail/partials/massive_action_selector', [
+            'surveyId'    => $surveyId,
+            'permissions' => $permissions
+        ], true);
+
+
+        $this->render('failedEmail_index', [
+            'failedEmailModel' => $failedEmailModel,
+            'pageSize'         => $pageSize,
+            'massiveAction'    => $massiveAction,
+        ]);
+    }
+
+    /**
+     * @throws CHttpException
+     */
+    public function actionResend()
+    {
+        $surveyId = sanitize_int(App()->request->getParam('surveyid'));
+        if (!$surveyId) {
+            throw new CHttpException(403, gT("Invalid survey ID"));
+        }
+        if (!Permission::model()->hasSurveyPermission($surveyId, 'responses', 'update')) {
+            App()->user->setFlash('error', gT("You do not have permission to access this page."));
+            $this->redirect(['failedemail/index/', 'surveyid' => $surveyId]);
+        }
+        $preserveResend = App()->request->getParam('preserveResend') ?? false;
+        $item = [App()->request->getParam('item')];
+        $items = json_decode(App()->request->getParam('sItems'));
+        $selectedItems = $items ?? $item;
+        if (!empty($selectedItems)) {
+            $criteria = new CDbCriteria();
+            $criteria->select = 'id, email_type, recipient';
+            $criteria->addCondition('surveyid', $surveyId);
+            $criteria->addInCondition('id', $selectedItems);
+            $failedEmails = FailedEmail::model()->findAll($criteria);
+            if (isset($failedEmails)) {
+                foreach ($failedEmails as $failedEmail) {
+                    $emailsByType[$failedEmail->email_type][$failedEmail->id] = $failedEmail->recipient;
+                }
+                if (isset($emailsByType)) {
+                    $result = sendSubmitNotifications($surveyId, $emailsByType, $preserveResend, true);
+                    if (!$preserveResend) {
+                        $criteria->addCondition('status', FailedEmail::STATE_SUCCESS);
+                        $failedEmails->deleteAll($criteria);
```

Comment:
$failedEmails is an array here. Should this be inside a loop? Or FailedEmails::modal()->deleteAll()?